YDotNet/Document/StickyIndexes/StickyIndex.cs

```diff
@@ -15,41 +14,33 @@ namespace YDotNet.Document.StickyIndexes;
 ///     Also, placing a sticky index at the end of a <see cref="Branch" /> will always point to the end of that
 ///     <see cref="Branch" />.
 /// </remarks>
-public class StickyIndex : IDisposable
+public class StickyIndex : UnmanagedResource
```

Comment:
Most handles should not be stored somewhere or only for debugging because the client does not have to cleanup any memory. So I made everything an unmanaged resource that needs to be disposed. I don't know how consistent it is right now and sometimes I have rhe bad habit to introduce too much base classes. So we can improve that. Sorry I am on mobile

Reply:
No problem about being on mobile, I could understand your message. 🙂

So, I just implemented the `UnmanagedResource` on `Doc` and `Branch` (which were the only classes I found with duplicate `nint Handle` properties). Now we only have a single place that declares a `Handle` property (except for other native representations like event payloads).

I also dropped the `TypeBase` and `ITypeBase` types. They were just "proxies" to be used in the `TypeCache` as far as I noticed, right? But since it's just an internal implementation detail and we can treat them as `UnmanagedResource` instances (since `TypeBase` was extending `UnamangedResource`), I did it to simplify the class structure.